org.eclipse.corrosion/src/org/eclipse/corrosion/RustManager.java

```diff
@@ -175,17 +176,20 @@ private static boolean runRustupCommand(SubMonitor monitor, String... arguments)
 			String[] command = new String[arguments.length + 1];
 			command[0] = rustup;
 			System.arraycopy(arguments, 0, command, 1, arguments.length);
-			ProcessBuilder builder = new ProcessBuilder(command);
-			builder.inheritIO();
-			Process process = builder.start();
+			Process process = new ProcessBuilder(command).start();
+			BufferedReader reader = new BufferedReader(new InputStreamReader(process.getErrorStream()));
+			CompletableFuture.runAsync(() -> {
+				reader.lines().forEachOrdered(line -> {
+					monitor.subTask(line);
+				});
+			});
 			while (process.isAlive() && !monitor.isCanceled()) {
```

Comment:
Same comment about implementing a wait loop while I believe Job cover this more easily.